<?xml version="1.0"?>

<launch>
	<arg name="robotName" value="bebop"/>
	<!-- SLOW -->
	<!--  arg name="kp" default="0.300"/>
	<arg name="ki" default="0.100"/>
	<arg name="kd" default="0.001"/>
	<arg name="alpha_p" default="0.1"/>
	<arg name="alpha_i" default="0.1"/>
	<arg name="alpha_d" default="0.5"/ -->
	<!-- FAST -->
	<arg name="kp" default="0.300"/>
	<arg name="ki" default="0.001"/>
	<arg name="kd" default="0.001"/>
	<arg name="alpha_p" default="0.1"/>
	<arg name="alpha_i" default="0.2"/>
	<arg name="alpha_d" default="0.4"/>

	<remap from="/uav/command_velocity" to="/$(arg robotName)/command_velocity"/>
	<remap from="/uav/command/motor_speed" to="/$(arg robotName)/command/motor_speed"/>
	<remap from="/uav/position" to="/$(arg robotName)/position"/>
	<remap from="/uav/velocity" to="/$(arg robotName)/linear_velocity"/>
	<remap from="/uav/noise/position" to="/$(arg robotName)/noise/position"/>
	<remap from="/uav/odometry" to="/$(arg robotName)/odometry"/>
	<remap from="/uav/true_odometry" to="/$(arg robotName)/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/$(arg robotName)/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/$(arg robotName)/trajectory_velocity"/>

	<node name="sit2flc" pkg="controllers" type="sit2flc" output="screen" args="$(arg kp) $(arg ki) $(arg kd) $(arg alpha_p) $(arg alpha_i) $(arg alpha_d)"/>
	<node name="trajectory" pkg="controllers" type="trajectory" output="screen" args=""/>
	<!-- node name="logger" pkg="controllers" type="logger" output="screen" args="1-fast-siit2fpid-$(arg alpha_p)-$(arg alpha_i)-$(arg alpha_d)"/ -->
	<node name="logger" pkg="controllers" type="logger" output="screen" args="4-fast-siit2fpid-noise-$(arg noise)"/>

	<!-- node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/ -->
</launch>
